src/frontend/src/optimizer/mod.rs

```diff
@@ -500,13 +500,42 @@ impl PlanRoot {
         Ok(plan)
     }
 
+    /// Auxiliary function to ensure there is **exchange directly before table scan**,
+    /// so that it will be pushed to compute node.
+    fn enforce_exchange_above_table_scan(plan: PlanRef) -> PlanRef {
+        if plan.node_type() != PlanNodeType::BatchExchange
+            && plan.node_type() != PlanNodeType::BatchSeqScan
+        {
+            let new_inputs = plan
+                .inputs()
+                .into_iter()
+                .map(|input| {
+                    if input.node_type() == PlanNodeType::BatchSeqScan
+                        && !has_batch_seq_scan_where(plan.clone(), |s| s.logical().is_sys_table())
```

Comment:
Doesn't seem like the best way to check if it is sys table scan. Feel free to suggest better alternatives.